json_serializable/lib/src/utils.dart

```diff
@@ -175,7 +175,19 @@ int _sortByLocation(FieldElement a, FieldElement b) {
 
 final _dartCoreObjectChecker = const TypeChecker.fromRuntime(Object);
 
-/// If a parameter is required to invoke the constructor,
+String genericClassArguments(ClassElement element, bool withConstraints) {
+  if (element.typeParameters.isEmpty) {
+    return '';
+  }
+  var values = element.typeParameters.map((t) {
+    if (withConstraints) {
+      return t.toString();
```

Comment:
consider ternary conditional when what you care about is the value rather than side effects

Reply:
ack